velox/exec/tests/HashJoinTest.cpp

```diff
@@ -6989,6 +6989,91 @@ DEBUG_ONLY_TEST_F(HashJoinTest, reclaimDuringTableBuild) {
       .run();
 }
 
+DEBUG_ONLY_TEST_F(HashJoinTest, exceptionDuringFinishJoinBuild) {
+  // This test is to make sure there is no memory leak when exceptions are
+  // thrown while parallelly preparing join table.
+  std::unique_ptr<memory::MemoryManager> memoryManager = createMemoryManager();
```

Comment:
Can we just use the default memory manager which has the shared memory arbitrator enabled? Thanks!